pkg/simple/client/monitoring/interface.go

```diff
@@ -1,39 +1,82 @@
 package monitoring
 
-
-type ClusterQuery struct {
-
+import "time"
+
+// MetricsLevel defines which level of metrics a Querier will retrieve.
+type MetricsLevel int
+
+const (
+	MetricsLevelCluster MetricsLevel = 1 << iota
+	MetricsLevelNode
+	MetricsLevelWorkspace
+	MetricsLevelNamespace
+	MetricsLevelWorkload
+	MetricsLevelPod
+	MetricsLevelContainer
+	MetricsLevelPvc
+	MetricsLevelComponent
+	MetricsLevelApp
+
+	StatusError          = "error"
+	StatusSuccess        = "success"
+	ErrorTypeUnavailable = "unavailable"
+	ErrorTypeInternal    = "internal"
+)
+
+// add identifier key for the sorting purpose
+const IdentifierKey = "resource_name"
+
+type MetricsGroup struct {
+	Results []Metrics `json:"results" description:"actual array of results"`
+
+	MetricsLevel string `json:"metrics_level" description:"metric level, eg. cluster"`
+	CurrentPage  int    `json:"page,omitempty" description:"current page returned"`
+	TotalPage    int    `json:"total_page,omitempty" description:"total number of pages"`
+	TotalItem    int    `json:"total_item,omitempty" description:"page size"`
 }
 
-type ClusterMetrics struct {
-
+type Metrics struct {
+	MetricsName string       `json:"metric_name,omitempty" description:"metric name, eg. scheduler_up_sum"`
+	Status      string       `json:"status" description:"result status, one of error, success"`
+	Data        *MetricsData `json:"data" description:"actual metric result"`
+	ErrorType   string       `json:"errorType,omitempty"`
+	Error       string       `json:"error,omitempty"`
 }
 
-type WorkspaceQuery struct {
-
-}
-
-type WorkspaceMetrics struct {
-
+type MetricsData struct {
+	MetricsType   string         `json:"resultType" description:"result type, one of matrix, vector"`
+	MetricsValues []MetricsValue `json:"result" description:"metric data including labels, time series and values"`
 }
 
-type NamespaceQuery struct {
-
-}
-
-type NamespaceMetrics struct {
+type Point [2]float64
 
+type MetricsValue struct {
+	Metadata map[string]string `json:"metric,omitempty" description:"time series labels"`
+	Sample   Point             `json:"value,omitempty" description:"time series, values of vector type"`
+	Series   []Point           `json:"values,omitempty" description:"time series, values of matrix type"`
 }
 
 // Interface defines all the abstract behaviors of monitoring
 type Interface interface {
+	GetMetrics(QueryOptions) *MetricsGroup
```

Comment:
Personally, I think `GetMetrics` is too general, now well designed,  and `QueryOptions` contains too much information. For example, if I want to query metrics of level `MetricsLevelWorkspace` , which fields of `QueryOptions` should I have to set, and how to set ? It's not obvious now.